### PR TITLE
[JUJU-2829] Prepare option to split docs into files.

### PR DESCRIPTION
**Pull request by juanmanuel-tirado**
This PR is an iteration in the markdown documentation generation. This PR addresses:
* `--split` flag to generate one markdown file per command
* `--url=<base url>` in case URLs link to a specific url
* The `see also` section has been moved to the top of the files following @tmihoc instructions

Example generated running with `--split --url=http://something` for the juju `add-k8s` command.

## See Also
[this](http://something/this)
[that](http://something/that)

## Summary
Adds a k8s endpoint and credential to Juju.

## Usage
```<k8s name>```

## Description

Creates a user-defined cloud based on a k8s cluster.

The new k8s cloud can then be used to bootstrap into, or it
can be added to an existing controller.

Use --controller option to add k8s cloud to a controller.
Use --client option to add k8s cloud to this client.

Specify a non default kubeconfig file location using $KUBECONFIG
environment variable or pipe in file content from stdin.

The config file can contain definitions for different k8s clusters,
use --cluster-name to pick which one to use.
It's also possible to select a context by name using --context-name.

When running add-k8s the underlying cloud/region hosting the cluster needs to be
detected to enable storage to be correctly configured. If the cloud/region cannot
be detected automatically, use either
  --cloud <cloudType|cloudName> to specify the host cloud
or
  --region <cloudType|cloudName>/<someregion> to specify the host
  cloud type and region.

Region is strictly necessary only when adding a k8s cluster to a JAAS controller.
When using a standalone Juju controller, usually just --cloud is required.

Once Juju is aware of the underlying cloud type, it looks for a suitably configured
storage class to provide operator and workload storage. If none is found, use
of the --storage option is required so that Juju will create a storage class
with the specified name.

If the cluster does not have a storage provisioning capability, use the
--skip-storage option to add the cluster without any workload storage configured.

Examples:

When your kubeconfig file is in the default location:
    juju add-k8s myk8scloud
    juju add-k8s myk8scloud --client
    juju add-k8s myk8scloud --controller mycontroller
    juju add-k8s --context-name mycontext myk8scloud
    juju add-k8s myk8scloud --region cloudNameOrCloudType/someregion
    juju add-k8s myk8scloud --cloud cloudNameOrCloudType
    juju add-k8s myk8scloud --cloud cloudNameOrCloudType --region=someregion
    juju add-k8s myk8scloud --cloud cloudNameOrCloudType --storage mystorageclass

To add a Kubernetes cloud using data from your kubeconfig file, when this file is not in the default location:
    KUBECONFIG=path-to-kubeconfig-file juju add-k8s myk8scloud --cluster-name=my_cluster_name

To add a Kubernetes cloud using data from kubectl, when your kubeconfig file is not in the default location:
    kubectl config view --raw | juju add-k8s myk8scloud --cluster-name=my_cluster_name

See also:
    remove-k8s

## Examples
`this example`
`this other example`

## Options
| Flag | Default | Usage |
| --- | --- | --- |
| `--B`, `--no-browser-login` | false | Do not use web browser for authentication |
| `--c`, `--controller` |  | Controller to operate in |
| `--client` | false | Client operation |
| `--cloud` |  | k8s cluster cloud |
| `--cluster-name` |  | Specify the k8s cluster to import |
| `--context-name` |  | Specify the k8s context to import |
| `--credential` |  | the credential to use when accessing the cluster |
| `--region` |  | k8s cluster region or cloud/region |
| `--skip-storage` | false | used when adding a cluster that doesn't have storage |
| `--storage` |  | k8s storage class for workload storage |

---

### CODE REVIEW

File: documentation.go [hunk 1]
```diff
@@ -138,16 +244,13 @@ func (c *documentationCommand) formatCommand(ref commandReference) string {
...
-		}
-		formatted += "\n"
+	// Options
+	formattedFlags := c.formatFlags(ref.command)
+	if len(formattedFlags) > 0 {
+		formatted += "## Options\n" + formattedFlags + "\n"
 	}
 
-	formatted += "---\n"
+	formatted += "---\n\n"
```
Comment (nvinuesa) [inline documentation.go:253]: why do we need this double newline?

File: aliasfile.go [hunk 2]
```diff
@@ -8,7 +8,7 @@ import (
 	"strings"
 )
 
-// ParseAliasFile will read the specified file and convert
+//  ParseAliasFile will read the specified file and convert
```
Comment (nvinuesa) [inline aliasfile.go:11]: small typo

**Review by nvinuesa — APPROVED** [review]
LGTM, thanks for this! I only added a small question (and to be sure that second newline is not an error)

File: documentation.go (hunk 1 above)
Comment (juanmanuel-tirado) [reply documentation.go:253]: In best practices ([here](https://www.markdownguide.org/basic-syntax#horizontal-rule-best-practices)), horizontal requires an empty line before other headings

File: aliasfile.go (hunk 2 above)
Comment (juanmanuel-tirado) [reply aliasfile.go:11]: Fixed.

**Pull request merged**